src/framebus.ts

```diff
@@ -102,8 +125,16 @@ export class Framebus {
     if (!payload) {
       return false;
     }
-
-    broadcast(window.top || window.self, payload, origin);
+    if (this.limitBroadcastToFramesArray) {
+      this.targetFrames.forEach((frame) => {
+        sendMessage(frame as Window, payload, origin);
+      });
+    } else {
+      broadcast(payload, {
+        origin,
+        frame: window.top || window.self,
+      });
+    }
```

Comment:
We need to use this helper function to convert the targetFrames to all be window objects. 

```suggestion
    if (this.limitBroadcastToFramesArray) {
      this.targetFramesAsWindows().forEach((frame) => {
        sendMessage(frame, payload, origin);
      });
    } else {
      broadcast(payload, {
        origin,
        frame: window.top || window.self,
      });
    }
```